ruby_02-web_applications_with_ruby/outlines/intro_to_crud_redux.markdown

```diff
@@ -0,0 +1,174 @@
+
+In our `index.erb` view:
+
+```erb
+<h1>All Tasks</h1>
+
+<% @tasks.each do |task| %>
+  <h3><a href="/tasks/<%= task.id %>"><%= task.title %></a></h3>
+  <p><%= task.description  %></p>
+  <a href="/tasks/<%= task.id %>/edit">Edit</a>
+<% end %>
+```
+
+In our controller:
+
+```ruby
+  get '/tasks/:id/edit' do
+    @task = Task.find(params[:id])
+    erb :edit
+  end
+```
+In our view, `edit.erb`:
+
+```erb
+<form action="/tasks/<%= @task.id %>" method="post">
+  <p>Edit</p>
+  <input type="hidden" name="_method" value="PUT" />
+  <input type='text' name='task[title]' value="<%= @task.title %>"/><br/>
+  <textarea name='task[description]'><%= @task.description %></textarea><br/>
+  <input type='submit'/>
+</form>
+```
+
+#### Updating a task
+
+Our new form needs somewhere to go when a user clicks submit. We'll use the **update** route to do the work of changing the task in the database.
+
+In our controller:
+
+```ruby
+  set :method_override, true  # this allows us to use _method in the form
+  ...
+  put '/tasks/:id' do |id|
+    Task.update(id.to_i, params[:task])
+    redirect "/tasks/#{id}"
+  end
+```
+
+In our TaskManager model:
+
+```ruby
+  def self.update(id, task_params)
+    database = SQLite3::Database.new('db/task_manager_development.db')
+    database.results_as_hash = true
+    database.execute("UPDATE tasks
+                      SET title = ?,
+                          description = ?
+                      WHERE id = ?;",
+                      task_params[:title],
+                      task_params[:description],
+                      id)
+
+    Task.find(id)
+  end
+```
+
+#### Deleting a task
+
+We don't need a form to delete a task, we just need to know which task we want to delete. We'll use a form to send a `DELETE` request to a route with our task id.
+
+In our `index.erb` view:
+
+```erb
+<% @tasks.each do |task| %>
+  <h3><a href="/tasks/<%= task.id %>"><%= task.title %></a></h3>
+  <p><%= task.description %></p>
+  <a href="/tasks/<%= task.id  %>/edit">Edit</a>
+  <form action="/tasks/<%= task.id %>" method="POST">
+    <input type="hidden" name="_method" value="DELETE">
+    <input type="submit" value="delete"/>
+  </form>
+<% end %>
+```
+
+In our controller:
+
+```ruby
+  delete '/tasks/:id' do |id|
+    task_manager.destroy(id.to_i)
+    redirect '/tasks'
+  end
+```
+
+In our TaskManager model:
+
+```ruby
+  def self.destroy(id)
+    database = SQLite3::Database.new('db/task_manager_development.db')
+    database.execute("DELETE FROM tasks
+                      WHERE id = ?;", id)
+  end
+```
+
+### Work Time
+
+#### Mild: Skill Inventory
+
+Fork [this repository](https://github.com/turingschool-examples/skill-inventory-crud-redux) and CRUD out a skill inventory. Users should be able to enter a skill (create), see a list of all of the skills, see each skill individually (read), edit a skill (update), and delete a skill (delete).
+
+#### Spicy: Robot World
```

Comment:
What makes this version more difficult? The dashboard piece? When thinking of "robots" vs "skills", robots are so much cooler. I wonder what we could do to differentiate these two more? Maybe this is a for later thought, but I want to mention it here to document it somewhere!